docs/migration.md

```diff
@@ -2,96 +2,253 @@
-pip install --pre -U "pydantic>=2.0a1"
+pip install --pre -U "pydantic>=2.0b1"
 ```
 
-!!! warning "Alpha release!"
-    Note that there are still some rough edges and incomplete features, and while trying out the Pydantic V2 alpha releases you may experience errors.
-
-    We encourage you to try out the alpha releases in a test environment and not in production.
-    Some features are still in development, and we will continue to make changes to the API.
+!!! warning "Beta release!"
+    Note that there are still some rough edges, and while trying out the Pydantic V2 beta releases you may experience errors.
+
+    We encourage you to try out the beta releases in a test environment and not in production.
+    While we do not currently intend to make changes to the API, it's still possible that we
+    will discover problems requiring breaking changes.
+
+If you do encounter any issues, please [create an issue in GitHub](https://github.com/pydantic/pydantic/issues) using
+the `bug V2` label. This will help us to actively monitor and track errors, and to continue to improve the library's
+performance.
+
+## Migration guide
+
+### Changes to `pydantic.BaseModel`
+
+Various method names have been changed; all non-deprecated `BaseModel` methods now have names matching either the
+format `model_.*` or `__.*pydantic.*__`. Where possible, we have retained the deprecated methods with their old names
+to help ease migration, but calling them will emit `DeprecationWarning`s.
+
+| Pydantic V2  | Pydantic V1 |
+| ------------ | ----------- |
+| `model_dump()` | `dict()` |
+| `model_dump_json()` | `json()` |
+| `model_copy()` | `copy()` |
+| `model_json_schema()` | `json_schema()` |
+| `model_construct()` | `construct()` |
+| `model_validate()` | `parse_obj()` |
+| `__pydantic_validator__` | `__validators__` |
+| `model_fields` | `__fields__` |
+| `__signature__` | `__signature__` |
+| `__class_vars__` | `__class_vars__` |
+| `__pydantic_private__` | `__private_attributes__` |
+
+* Some of the built-in data-loading functionality has been slated for removal. In particular,
+    `parse_raw` and `parse_file` are now deprecated. In Prefect V2, `model_validate_json` works like `parse_raw`. Otherwise, you should load the data and then pass it to `model_validate`.
+* The `from_orm` method has been removed; you can now just use `model_validate` (equivalent to `parse_obj` from
+  Pydantic V1) to achieve something similar, as long as you've set `from_attributes=True` in the model config.
+* The `__eq__` method has changed for models.
+    * Models can only be equal to other `BaseModel` instances.
+    * For two model instances to be equal, they must have the same:
+        * Type (or, in the case of generic models, non-parametrized generic origin type)
+        * Field values
+        * Extra values (only relevant when `model_config['extra'] == 'allow'`)
+        * Private attribute values; models with different values of private attributes are no longer equal.
```

Comment:
Does this still applies with the latest changes on `PrivateAttr`? Just asking.

Reply:
@dmontagu could you address this comment? I've taken care of the rest